### ISSUE TITLE
(-215:Assertion failed) inputs.size() in function cv::dnn::dnn4_v20191202::Layer::getMemoryShapes

### ISSUE THREAD

**Issue by BadMachine** [opencv/opencv#16323]
##### System information (version)

- OpenCV => 4.2.0-pre
- Operating System / Platform => Windows 10 64 Bit
- Compiler => Visual Studio 2016

##### Detailed description

When im trying to load ONNX model im getting error:

F:\\OpenCV\\opencv-master\\modules\\dnn\\src\\dnn.cpp:4479: error: (-215:Assertion failed) inputs.size() in function 'cv::dnn::dnn4_v20191202::Layer::getMemoryShapes'\n" ...}

#### Model architecture:
HRNet
#### Model was exported from PyTorch to ONNX with following code:
```C++
    model.load_state_dict(torch.load(args.model_file))
    data = torch.zeros((1, 3, 288, 384)).cuda()
    model.eval()
    model.cuda()
    model.float()

    torch.onnx.export(model, data, 'test.onnx')
```
#### Model properties and first input layer:
![image](https://user-images.githubusercontent.com/50546204/72201532-275df900-3466-11ea-8e91-7a9e2d8a85a5.png)

#### About error
Looks like error appears in onnx_importer.cpp at line
```C++
        Ptr<Layer> layer = dstNet.getLayer(id);
        layer->getMemoryShapes(layerInpShapes, 0, layerOutShapes, layerInternalShapes);
```
on layer id = 113

##### Steps to reproduce:

1. [Upload_model](https://drive.google.com/open?id=1qQ5oFBk8UjzK0FAYLBaCYcGYfKMm5AXB)
2. load in:
```C++
auto network = readNetFromONNX("test.onnx");
```

**Comment by pranavpandey2511** [comment]
@BadMachine Hi, I was also trying to do the same thing last and was facing the same issue. I exported the pytorch the pytorch model to onnx and tried to load it in opencv dnn modules and was getting error in getmemoryshape() function.

I figured out that it was a mistake from my end only, before converting the model to Onnx format, **you have to put your pytorch model in eval mode using model.eval()** and then convert it to onnx. It worked right away when I loaded it in opencv after doing this.

Hope this solves your problem too.

**Comment by BadMachine** [comment]
@pranavpandey2511 
Thanks for your suggestion, but my model already in eval mode. So, it doesnt work anyway.

**Comment by ashishkrshrivastava** [comment]
@dkurt, I found the problem in layer `111` which is somewhere here. 
![Screenshot from 2020-02-08 09-42-55](https://user-images.githubusercontent.com/26292819/74079017-7c862e00-4a57-11ea-8af3-5593bff81f1b.png)
I think issue [16475](https://github.com/opencv/opencv/issues/16475) has the similar problem.

**Comment by ashishkrshrivastava** [comment]
@BadMachine Can you please export and upload model using Pytorch 1.3 instead of Pytorch 1.2.
